plutus-playground-client/src/MainFrame.purs

```diff
@@ -381,6 +372,29 @@ eval (PopulateAction n l event) = do
     (evalForm event)
   pure $ extract event
 
+evalWalletEvent :: WalletEvent -> Array SimulatorWallet -> Array SimulatorWallet
+evalWalletEvent AddWallet wallets =
+  let maxWalletId = fromMaybe 0 $ maximumOf (traversed <<< _simulatorWalletWallet <<< _walletId) wallets
+      newWallet = mkSimulatorWallet (maxWalletId + 1)
+  in Array.snoc wallets newWallet
```

Comment:
not sure why but it feels like it shouldn't be necessary to traverse this data structure like this in order to get the next wallet id.

Reply:
Imagine you add 5 wallets, then delete the first four. You now have one wallet, and it's number 5. So, just `wallets.length + 1` won't do it.

We could sort descending by ID and take the ID of the first one, but I'm not sure that's better than this.

We could maintain a separate counter, but that's duplicate state that's just asking to be out of sync.

Erm...unless you see something I'm missing, I stand by this. :-)

Reply:
pitty there isn't a typeclass for things that can be `inc`ed, then you could have something like `Incable k => Map k v` with an operation `insert :: v -> Map k v -> (k, Map k v)`

Anyway, I didn't have a solution, just that it felt like something is missing and that this idea, similar to a primary key, should be available in a ready made data structure somewhere.

Reply:
Interesting idea. FWIW `Semiring` gives you `add` and `one`: https://pursuit.purescript.org/packages/purescript-prelude/4.1.0/docs/Data.Semiring#v:add